### DISCUSSION TITLE
BRouter start on roundabout

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#150] [Q&A] [answered]
I have just tried a few things and it boils down to roundabouts and slip-roads as starting points - replicated in several places on a map using duplicate "preferences" on both devices.  If you attempt to start from a roundabout (ie: a place where fuel/services may be) then it won't "route".  The on/off-ramp/slip-road does not route, either.  You have to start the Route on the main carriageway or a little back from the roundabout.

See example below:
It will "route" on the main A15 towards the bridge (Humber Bridge), but not on the entry-ramp, so you have to fine a "clear" piece of road.

![cruiser-brouter-issue](https://user-images.githubusercontent.com/125696109/235621409-e78845c3-0bbc-4d13-a0be-5574933720ef.jpg)

Given that this issue would affect any user of BRouter+Cruiser, certainly worth posting here, too :)

_Originally posted by @MotoUKRider in https://github.com/devemux86/cruiser/discussions/149#discussioncomment-5778464_

**Comment by devemux86** [comment]
@MotoUKRider @0709wiwiwi 

The issue is in Cruiser reader of the BRouter responses. The 1st `trkpt` contains:
```xml
<sym>roundabout_e2</sym>
<type>Via</type>
```

Usually we expect in the 1st `trkpt`:
```xml
<sym>pass_place</sym>
<type>Via</type>
```

Looks like we need to adjust the Cruiser reader of BRouter responses?

---

BRouter issue:
- https://github.com/abrensch/brouter/issues/545

**Reply by MotoUKRider** [reply]
I'm so glad it wasn't me, but frustrating that I wasn't able to see this myself via debugging in some way :)  ie: "No route" is quite vague.

It's nice to be able to pick out little things like this, and it could certainly help others if they have Routes that start in a similar way.

With my previous experience of items (such as level-crossings) being tagged in different ways preventing pass, I had thought that it may have been the (free to motorcycles) Humber Bridge toll, then perhaps the roundabout itself, but then when I tried other roundabouts in random places, I experienced the same issue.

**Comment by devemux86** [comment]
@MotoUKRider thanks for your time and tests! :+1: 

We need to clarify with @afischerdev and then proceed with the fix. :slightly_smiling_face:

**Comment by devemux86** [comment]
@MotoUKRider thanks for the report!

It will work properly in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
It works properly in [Cruiser 3.0.19](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by MotoUKRider** [reply]
Thanks - tested and working with latest Cruiser+BRouter combo :)
